### PR TITLE
Add Github workflows for unit tests, go tidiness and code linters

### PR DESCRIPTION
**Pull request by antoninbas**
The static code checkers revealed a lot of issues, some of them quite
"interesting". I fixed them, even though there is still room for
improvement in terms of code quality.

Signed-off-by: Antonin Bas <abas@vmware.com>

### CODE REVIEW

File: protocol/dhcp.go [hunk 1]
```diff
@@ -74,9 +80,20 @@ const (
 //	FLAG_BROADCAST_MASK uint16 = (1 << FLAG_BROADCAST)
 )
 
+func getRandomXID() (uint32, error) {
```
Comment (GraysonWu) [inline protocol/dhcp.go:83]: What's the advantage of using this way to generate a random uint32?

**Review by GraysonWu — COMMENTED** [review]
Thanks for fixing these @antoninbas . And it's really good that we finally have our libOpenflow fork. I will fix similar problems in my future dev if I run into them. Changed LGTM overall and I left a question comment in code.

File: protocol/dhcp.go (hunk 1 above)
Comment (antoninbas) [reply protocol/dhcp.go:83]: It's a cryptographic secure random number generator. Not sure it really matters here (it's all about someone's ability to predict the xid), but it avoids linter warnings and it doesn't make the code too complicated.

Comment (GraysonWu) [reply protocol/dhcp.go:83]: Got it. Thanks!

**Review by GraysonWu — APPROVED** [review]

**Pull request merged**